node/request.js

```diff
@@ -188,6 +190,29 @@ TChannelRequest.prototype.resend = function resend() {
     }
 };
 
+TChannelRequest.prototype.checkTimeout = function checkTimeout(err, res) {
+    var self = this;
+    var now = self.timers.now();
+    self.elapsed = now - self.start;
+    if (self.elapsed < self.timeout) return false;
+    if (err) {
+        if (!self.err) {
+            self.emit('error', err);
+        }
+    } else if (res) {
+        if (!self.err && !self.res) {
+            self.emit('response', res);
```

Comment:
No, that's done by `TChannelRequest#onResponse` which gets hooked up by constructor so that we don't have to do that in every emission site.